recipes/install.rb

```diff
@@ -70,11 +72,12 @@
 end
 
 # Create keyFile if specified
-file node['mongodb']['config']['keyFile'] do
-  owner node['mongodb']['user']
-  group node['mongodb']['group']
-  mode  '0600'
-  backup false
-  content node['mongodb']['key_file_content']
-  only_if { node['mongodb']['key_file_content'] }
+if node['mongodb']['key_file_content']
+  file node['mongodb']['config']['keyFile'] do
+    owner node['mongodb']['user']
+    group node['mongodb']['group']
+    mode  '0600'
+    backup false
+    content node['mongodb']['key_file_content']
+  end
```

Comment:
OK. Just to clarify that, after several tests, I agree with the hack :+1: A better solution would require more work and it falls out of the scope of this PR.